### ISSUE TITLE
Can Payum be used as part of a REST API?

### ISSUE THREAD

**Issue by PicassoHouessou** [Payum/Payum#946]
I tried and integrate PayumBundle in a Symfony project and it worked. My problem is that I'm building a REST API and I have the frontend in React. I wonder if it is possible to use Payum in this case. If yes how to proceed?

**Comment by webard** [comment]
Yes, in most of cases just return in your API redirect url and let this redirect handle in your frontend app.

**Comment by pierredup** [comment]
For payments that redirect to the gateway, it should work fine in a REST API. As @webard mentioned, you can just return the redirect URL in the JSON response, then the frontend app can redirect to the URL. Although returning the response might be a bit more tricky. You would need to set the return URL to a page in your app, which then calls an endpoint, passing all the parameters, to verify and process the payment, and then return the proper response back to the app. There are some security considerations as well to take into account with this.

For payments that uses [HttpPostRedirect](https://github.com/Payum/Payum/blob/366d4f7dacd3a71c1ec5cf8886143ed6bfe8abe7/src/Payum/Core/Reply/HttpPostRedirect.php), you would need to pass all the fields back to the frontend, then do a POST request to the payment gateway.

Then there are other gateways that just renders a form, so there is no redirect involved. You can manually construct the form on the frontend, although you would need to do so for every payment gateway that you want to implement, since each one can have a different form (E.G a form capturing credit card information vs a form requiring other details). An alternative is to render the form on the server-side, using the existing templates that Payum provides, and just updating your app to render the returned HTML.

An [example capture](https://github.com/Payum/Payum/blob/ffa0fb8c04967d7d1e1f8bb122f5e4123b06fe5e/docs/examples/capture-script.md#secured-script) is available, but instead of outputting the response and headers, you would just return those values as part of a JSON response, and handle them from whichever application is making the request (again, there might be some security considerations with this approach).

TL;DR: It's possible to use Payum as part of a REST API, but there is some bit of manual work involved.
